src/Activities/Internal/ActivityMetricsSender.cs

```diff
@@ -18,17 +19,31 @@ internal sealed class ActivityMetricsSender : IActivitiesEventSender, IDisposabl
 		private readonly Meter m_meter;
 		private readonly Counter<double> m_activityCounter;
 		private readonly Counter<double> m_healthCheckActivityCounter;
+		private readonly Histogram<double> m_activityHistogram;
+		private readonly Histogram<double> m_healthCheckActivityHistogram;
+		private readonly bool m_useHistogramForActivity;
 		private readonly IExecutionContext m_context;
 		private readonly IHostEnvironment m_hostEnvironment;
 		private readonly ArrayPool<KeyValuePair<string, object?>> m_arrayPool;
 
-		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment)
+		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment, IConfiguration configuration)
```

Comment:
I would advise against using `IConfiguration` directly please use `IOptions` pattern instead to inject only the section you are interested in and avoid doing parsing manually. https://learn.microsoft.com/en-us/dotnet/core/extensions/options

Reply:
You need to use the following method to register the option:
```Csharp
collection
	.AddOptions<ServiceProviderOptions>()
	.BindConfiguration(UseHistogramForActivityMonitoringConfigurationPath);
```
It would bind it to the configuration obtained from DI without you actually having an instance.